### ISSUE TITLE
Network id configuration doesn't protect user from deploying to different network

### ISSUE THREAD

**Issue by tcoulter** [trufflesuite/truffle#1316]
As we saw during the retreat demo, if you specify a specific number in your truffle config (say, `3`), if the actual network id of the chain doesn't match, Truffle won't error (and it should, to protect the user from inadvertently deploying with real Ether). 

Steps to repro:
1. Set Truffle config network id to 3.
1. Deploy to rinkeby (which has network id of 4). Notice Truffle doesn't error, and performs the deployment.

Note: This bug report was written very quickly because we're at the retreat demo. Feel free to ask me any questions if the above is unclear.

**Comment by gnidan** [comment]
Not stale

**Comment by gnidan** [comment]
Colony's build is [failing](https://circleci.com/gh/JoinColony/colonyNetwork/5323?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link), seems to be a result of the change in the above PR.

Relevant CI output:
```
...
Summary
=======
> Total deployments:   13
> Final cost:          0 ETH

Error: [object Object]
    at detectNetworkId (/home/circleci/colonyNetwork/node_modules/truffle/build/webpack:/packages/truffle-core/lib/environment.js:73:1)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:182:7)
Truffle v5.0.0-next.20 (core: 5.0.0-beta.2)
```

Looks like the error is constructed wrong [here](https://github.com/trufflesuite/truffle/blob/next/packages/truffle-core/lib/environment.js#L65-L74), although I might just be blanking on whether `{ message: "..." }` is kosher.

Hard to say if there's a real problem, we may have caught a mismatch in the test suite :)

cc @elenadimitrova

**Comment by gnidan** [comment]
This is done and released in v5.0.0. Let us know if you run into it again! Thanks!
